tests/network/sriov.go

```diff
@@ -682,7 +682,8 @@ func defaultCloudInitNetworkData() string {
 func findIfaceByMAC(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance, mac string, timeout time.Duration) (string, error) {
 	var ifaceName string
 	err := wait.Poll(timeout, 5*time.Second, func() (bool, error) {
-		vmi, err := virtClient.VirtualMachineInstance(vmi.GetNamespace()).Get(vmi.GetName(), &k8smetav1.GetOptions{})
+		var err error
+		vmi, err = virtClient.VirtualMachineInstance(vmi.GetNamespace()).Get(vmi.GetName(), &k8smetav1.GetOptions{})
```

Comment:
I am not sure what you mean here.
The pointer is updated, the value of the original one is not.

The pointer is suppose to be updated, as it is used in the loop. I think it is fine.

Reply:
I mean the `findIfaceByMAC` will mutate the vmi it is passing it (it will feature an updated VMI w/ a different status). 

Code should be side-effect free - if you don't want to shadow the variable name, just use another variable within the poll loop.

Reply:
I am claiming that it will not mutate the `vmi` variable.
Same as a string is immutable, if we had a keyword in the language to say that `vmi` is immutable, it would have worked.

The `vmi` content is not mutated, once the assignment here takes place, it replaces the original pointer to something else and will not touch the content.

I think this is fine, we use this all over already.
If you are trying to claim that we should never re-use the variables that are passing in as arguments, I would like an explanation on why not. In this case, it seems natural.

Reply:
I think the Go idiomatic way to say something like this is to shadow it at the beginning: `vmi := vmi`.
I personally do not think it is needed, it feels like a waste of time, especially that if the members are of type map/slice/pointer, they still will point to the original data and you may mutate it if you mutate the shadowed variable (and the renamed one if you wish).

Bottom line, if you want to make sure you do not mutate the argument that gets in, your logic is not enough, you need to make sure it is "deep copied". Go is lacking this mutation check, which is sad but there is nothing we can do about it.

Reply:
And I am trying to argue that the fact that I will use a different name will *not* protect against the point you raised.
You are claiming that using a different variable will make a difference, and I am telling you it will not make any difference, not here and not in a general sense.

If I review changes, and the variable is changed or not changed, makes no difference to me.
I cannot determine if it will get mutated or not based on this rule you gave here.

Reply:
My comment is about more than this example: it's about good practices. I honestly don't see what good can come out of re-defining an input variable - and as such, I avoid it, and try to steer people away from it. You want to do it ? Go ahead. I think the code actually looks worse now than it did before, and as such, I cannot lgtm it. It literally does *not* look good to me. 

... it's also paradoxical to see you introduce a commit about `tests, sriov: Do not shadow the 'vmi' variable` while advocating " the Go idiomatic way to say something like this is to shadow" the variable.

Reply:
> My comment is about more than this example: it's about good practices. I honestly don't see what good can come out of re-defining an input variable - and as such, I avoid it, and try to steer people away from it. You want to do it ? Go ahead. I think the code actually looks worse now than it did before, and as such, I cannot lgtm it. It literally does not look good to me.

You just not managed to convinced me this practice is one we should follow. I answered your reasoning which you raised, hoping we can focus the discussion there.
I usually prefer not to use this as an argument, but because there is no other reasoning from your side, I could just throw it here: I have found no such practice recommended in any Go writing, nor I noticed any linter trying to aim there.

I will be more then happy to see a convincing argument on why using this is not a good practice.

> ... it's also paradoxical to see you introduce a commit about tests, sriov: Do not shadow the 'vmi' variable while advocating " the Go idiomatic way to say something like this is to shadow" the variable.

I think you misunderstood me here.
**If** one wants to avoid the usage of a variable, i.e. protect it, it can be shadowed.
But, I do not think we need to protect it in this case, therefore I see no reason to use shadowing or a different variable.

IMO the code that is here is making things harder to understand. There is a loop, at the beginning it uses the argument from the input, then it gets shadowed and no longer use it... this is implicit and has a cognitive load (for me at least).
I wanted to remove this, as it is not useful.

Reply:
I don't want to hold this PR hostage; if you move the commit into another dedicated PR (since that commit has nothing to do with the fix you're reporting in this one) I would gladly LGTM it.

Reply:
I support @maiqueb's claim that it requires some mental energy to understand what is going on here.
@EdDev please re-consider changing the `vmi` variable inside the closure.

Reply:
in case there is an error, it overrides the vmi with invalid struct, we can lose the vmi.name this way i think

Reply:
> this is just incorrect. The original vmi value will never change.

I saw it happens, for example when there is etcd timeout, it is also on the docs, that it can override it in case of an error

Reply:
@EdDev you don't have to keep shadowing.
IMO this small name change can reduce the mental burden:
```go
fetchedVmi, err := virtClient.VirtualMachineInstance(vmi.GetNamespace()).Get(vmi.GetName(), &k8smetav1.GetOptions{})
```

This way, the outer `vmi` is only referenced twice in the context of getting its namespace and name.

Reply:
If I’ll need to do anything, is to not change it at all and keep the existing.

I find it wrong to create a new variable, same as shadowing.

it is also not backed by existing usages all over the code, which makes sense.. it is not needed.

Reply:
> I see no loop in the function - nor I see the argument from the input being used. I see a poll operation in which the first thing done is to shadow the variable, something to me actually makes sense.

Right, it is worse. The original vmi variable is binded in the closure, which means we query about the vmi through the “old” values. In this case it does not matter, but in other cases it does.

Reply:
There is another alternative: get the vmi's namespace and name from outside of the closure.

Reply:
This sounds like a good compromise, I kind of line it. It gives a good reason for using a different variable.
I will give it a try.

Reply:
Well, it does not solve the disagreement. It just solves the closure and then I need to use a different variable name (or shadow) just because you do not want me to use the one that was passed in as an argument.

Reply:
I dropped the commit that removed the shadowing.
I think we need to drop it as I did, but it is wasteful to block the whole change on this little thing.

Done